<launch>
  <arg name="linear_throttle_factor" value="1.0" />
  <arg name="angular_throttle_factor" value="1.0" />
  <arg name="finger_throttle_factor" value="1.0" />
  
  <node pkg="joy" type="joy_node" name="joy_node">
    <param name="dev" type="string" value="/dev/input/js1" />
    <param name="deadzone" type="double" value="0.033" />
    <param name="autorepeat_rate" type="double" value="30" />
    <param name="coalesce_interval" type="double" value="0.033" />
  </node>
  
  <node name="kinova_teleop" pkg="kinova_teleop" type="kinova_teleop" output="screen">
    <param name="linear_throttle_factor" type="double" value="$(arg linear_throttle_factor)" />
    <param name="angular_throttle_factor" type="double" value="$(arg angular_throttle_factor)" />
    <param name="finger_throttle_factor" type="double" value="$(arg finger_throttle_factor)" />
    <remap from="/joint_velocity" to="/j2n6a300_driver/in/full_joint_velocity"/>
    <remap from="/cartesian_velocity" to="/j2n6a300_driver/in/full_cartesian_velocity"/>
    <remap from="/stop" to="/j2n6a300_driver/in/stop" />
    <remap from="/start" to="/j2n6a300_driver/in/start" />
    <remap from="/home_arm" to="/j2n6a300_driver/in/home_arm" />
  </node>

</launch>
